app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,34 @@
 class PostsController < ApplicationController
+  before_action :current_user, only: [:create]
   def index
     @user = User.find(params[:user_id])
-    @posts_list = @user.recent_posts
+    @posts_list = @user.recent_3_posts
   end
 
   def show
     @user = User.find(params[:user_id])
-    @post = Post.find(paraams[:id])
+    @post = Post.find(params[:id])
+  end
+
+  private
```

Comment:
You need to make only need to make your `post_params` method private not the `create` action